Global Jest setup/teardown with NX libraries does not find modules

I am trying to run `globalSetup` and `globalTeardown` functions for Jest in my NX-15 typescript workspace. 

Everything is vanilla, meaning I did not change any of the library setups. Everything was generated using `nx g @nrwl/js:library ...`. Workspace was setup using the `--preset=ts` flag.

### Current Behavior

The current behavior is that project module imports for `globalSetup` libraries simply do not work, no matter what I try.

### Expected Behavior

When setting up a workspace with `--preset=ts` flags and using the default Jest setup as generated by `@nrwl/js`, running `globalSetup` functions should work without additional config. At a minimum, there should be workable config examples in the docs.

### Github Repo

https://github.com/wolframm/nx-jest-globalSetup

### Steps to Reproduce

1) Setup workspace and some libraries (done in Git repo above)
```
npx create-nx-workspace@latest myorg --preset=ts
nx generate @nrwl/js:library --name=someLib --compiler=swc --includeBabelRc
nx generate @nrwl/js:library --name=someTestLib --compiler=swc --includeBabelRc
nx generate @nrwl/js:library --name=someSetupLib --compiler=swc --includeBabelRc
```

2) Modify `myorg/jest.preset.js`:

```
const nxPreset = require("@nrwl/jest/preset").default;
const path = require("path");

/** @type {import("@jest/types").Config.InitialOptions} */
module.exports = {
  ...nxPreset,
  globalSetup: path.join(__dirname, "jest-global-setup.ts")
};
```

3) Create `myorg/jest-global-setup.ts` as per https://nx.dev/packages/jest:

```
import { registerTsProject } from 'nx/src/utils/register';
const cleanupRegisteredPaths = registerTsProject('.', 'tsconfig.base.json');

import { someSetupLib } from '@myorg/some-setup-lib';
export default async function () {
  someSetupLib();
}
// make sure to run the clean up!
cleanupRegisteredPaths();
```

4) Import something into `myorg/packages/some-setup-lib/src/lib/some-setup-lib.ts`:

```
import {someLib} from "@myorg/some-lib";

export function someSetupLib(): string {
  console.log(someLib());
  return 'some-setup-lib';
}
```

5) Run a test: ` nx run some-test-lib:test`

### Failure Logs

```shell
/usr/local/bin/npx nx run some-test-lib:test

> nx run some-test-lib:test

Determining test suites to run...
 >  NX   Jest: Got error running globalSetup - /Users/arnewolframm/myorg/jest-global-setup.ts, reason: Cannot find module '@myorg/some-setup-lib'

   Require stack:
   - /Users/arnewolframm/myorg/jest-global-setup.ts
   - /Users/arnewolframm/myorg/node_modules/@jest/transform/node_modules/jest-util/build/requireOrImportModule.js
   - /Users/arnewolframm/myorg/node_modules/@jest/transform/node_modules/jest-util/build/index.js
   - /Users/arnewolframm/myorg/node_modules/@jest/transform/build/ScriptTransformer.js
   - /Users/arnewolframm/myorg/node_modules/@jest/transform/build/index.js
   - /Users/arnewolframm/myorg/node_modules/jest-runtime/build/index.js
   - /Users/arnewolframm/myorg/node_modules/@jest/core/build/cli/index.js
   - /Users/arnewolframm/myorg/node_modules/@jest/core/build/index.js
   - /Users/arnewolframm/myorg/node_modules/jest/build/index.js
   - /Users/arnewolframm/myorg/node_modules/@nrwl/jest/src/executors/jest/jest.impl.js
   - /Users/arnewolframm/myorg/node_modules/nx/src/config/workspaces.js
   - /Users/arnewolframm/myorg/node_modules/nx/src/command-line/run.js
   - /Users/arnewolframm/myorg/node_modules/nx/bin/run-executor.js
   Pass --verbose to see the stacktrace.


 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target test for project some-test-lib (2s)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
 

Process finished with exit code 1
```

### Nx Report

```shell
Node : 16.17.1
   OS   : darwin x64
   npm  : 8.19.2
   
   nx : 15.2.1
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.2.1
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.2.1
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.2.1
   @nrwl/js : 15.2.1
   @nrwl/linter : 15.2.1
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : Not Found
   @nrwl/workspace : 15.2.1
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

--- a/docs/generated/packages/jest/documents/overview.md
+++ b/docs/generated/packages/jest/documents/overview.md
@@ -94,7 +94,7 @@ This is because each [jest process creates a workers based on system resources](
 
 ### Jest
 
-Primary configurations for Jest will be via the `jest.config.js` file that generated for your project. This file will extend the root `jest.config.js` file. Learn more about [Jest configurations](https://jestjs.io/docs/configuration#options).
+Primary configurations for Jest will be via the `jest.config.ts` file that generated for your project. This file will extend the root `jest.preset.js` file. Learn more about [Jest configurations](https://jestjs.io/docs/configuration#options).
 
 ### Nx
 
@@ -119,7 +119,7 @@ By default, coverage reports will be generated in the `coverage/` directory unde
 In order to use Jest's global setup/teardown functions that reference nx libraries, you'll need to register the TS path for jest to resolve the libraries.
 Nx provides a helper function that you can import within your setup/teardown file.
 
-```typescript
+```typescript {% fileName="global-setup.ts" %}
 import { registerTsProject } from 'nx/src/utils/register';
 const cleanupRegisteredPaths = registerTsProject('.', 'tsconfig.base.json');
 
@@ -130,6 +130,23 @@ export default async function () {
 // make sure to run the clean up!
 cleanupRegisteredPaths();
 ```
+
+{% callout type="note" title="@swc/jest & global scripts" %}
+When using @swc/jest and a global setup/teardown file,
+You'll have to set the global setup/teardown file to be transformed with ts-jest.
+For example, if your files are named `global-setup.ts` and `global-teardown.ts`,
+then you would need to add to your _project level `jest.config.ts`_ a new entry in the transformers object
+
+```typescript {% fileName="apps/<your-project>/jest.config.ts" %}
+export default {
+  transform: {
+    'global-(setup|teardown).ts': 'ts-jest',
+    // resest of the transformers
+  },
+};
+```
+
+{% /callout %}
 
 ## Debugging Failing Tests
 

--- a/docs/shared/packages/jest/jest-plugin.md
+++ b/docs/shared/packages/jest/jest-plugin.md
@@ -94,7 +94,7 @@ This is because each [jest process creates a workers based on system resources](
 
 ### Jest
 
-Primary configurations for Jest will be via the `jest.config.js` file that generated for your project. This file will extend the root `jest.config.js` file. Learn more about [Jest configurations](https://jestjs.io/docs/configuration#options).
+Primary configurations for Jest will be via the `jest.config.ts` file that generated for your project. This file will extend the root `jest.preset.js` file. Learn more about [Jest configurations](https://jestjs.io/docs/configuration#options).
 
 ### Nx
 
@@ -119,7 +119,7 @@ By default, coverage reports will be generated in the `coverage/` directory unde
 In order to use Jest's global setup/teardown functions that reference nx libraries, you'll need to register the TS path for jest to resolve the libraries.
 Nx provides a helper function that you can import within your setup/teardown file.
 
-```typescript
+```typescript {% fileName="global-setup.ts" %}
 import { registerTsProject } from 'nx/src/utils/register';
 const cleanupRegisteredPaths = registerTsProject('.', 'tsconfig.base.json');
 
@@ -130,6 +130,23 @@ export default async function () {
 // make sure to run the clean up!
 cleanupRegisteredPaths();
 ```
+
+{% callout type="note" title="@swc/jest & global scripts" %}
+When using @swc/jest and a global setup/teardown file,
+You'll have to set the global setup/teardown file to be transformed with ts-jest.
+For example, if your files are named `global-setup.ts` and `global-teardown.ts`,
+then you would need to add to your _project level `jest.config.ts`_ a new entry in the transformers object
+
+```typescript {% fileName="apps/<your-project>/jest.config.ts" %}
+export default {
+  transform: {
+    'global-(setup|teardown).ts': 'ts-jest',
+    // resest of the transformers
+  },
+};
+```
+
+{% /callout %}
 
 ## Debugging Failing Tests
 

